src/cryptonote_core/blockchain_based_list.cpp

```diff
@@ -149,21 +118,41 @@ void BlockchainBasedList::apply_block(uint64_t block_height, const crypto::hash&
       current_supernodes.emplace_back(std::move(sn));
     }
 
+      //seed RNG
+
+    std::seed_seq seed(reinterpret_cast<const unsigned char*>(&block_hash.data[0]),
+                       reinterpret_cast<const unsigned char*>(&block_hash.data[sizeof block_hash.data]));
+
+    m_rng.seed(seed);
+
       //sort valid supernodes by the age of stake
 
     std::sort(current_supernodes.begin(), current_supernodes.end(), [](const supernode& s1, const supernode& s2) { return s1.block_height < s2.block_height; });
```

Comment:
The sorting here is not quite deterministic -- it does not totally define ordering among supernodes registered in the same block.  A fallback to comparison on the id when the heights are equal would fix it:

```C++
return s1.block_height < s2.block_height ||
    (s1.block_height == s2.block_height && s1.supernode_public_id < s2.supernode_public_id);
```